File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
class Solution {
    public int[] solution(int[] arr) {
        if(arr.length==1) {
            int[] answer = {-1};
            return answer;
        }
        else{
            int min = arr[0];
            int idx = 0;
            for(int i=0; i<arr.length; i++){
                min = Math.min(min,arr[i]);
            }
            int[] answer = new int[arr.length-1];
            for(int i=0; i<arr.length; i++){
                if(min==arr[i]) continue;
                answer[idx++]=arr[i];
            }
            return answer;
        }
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    answer.push_back(-1);
    int min_idx = min_element(arr.begin(), arr.end())-arr.begin();
    arr.erase(arr.begin()+min_idx);
    if(arr.size()>0) return arr;
    else return answer;
}